src/responder/common/cache_req/plugins/cache_req_svc_by_port.c

```diff
@@ -89,8 +89,20 @@ cache_req_svc_by_port_lookup(TALLOC_CTX *mem_ctx,
                              struct sss_domain_info *domain,
                              struct ldb_result **_result)
 {
-    return sysdb_getservbyport(mem_ctx, domain, data->svc.port,
-                               data->svc.protocol.lookup, _result);
+    struct ldb_result *res = NULL;
+    errno_t ret;
+
+    ret = sysdb_getservbyport(mem_ctx, domain, data->svc.port,
+                              data->svc.protocol.lookup, &res);
+
+    /* RFC6335 Section 5 allows more than one service associated with a
+     * particular transport protocol and port. In such case return only
+     * the first result. */
+    if (ret == EOK && res->count > 1) {
+        res->count = 1;
+    }
+
```

Comment:
Shouldn't `res` be assigned to `_result`? Why using the intermediate `res` variable?

Reply:
Yes, and it was not caught because sysdb-tests just test the DB layer and the integration tests lack tests for service lookups. I have fixed that.